### PR TITLE
Docker for Mac: NFS

### PR DESCRIPTION
**Pull request by tripox**
1. Documentation for initial NFS setup on macOS
2. Example `Makefile`
3. `docker-compose-mac.yml` file for macOS
4. Using volumes in `docker-compose.yml` for the root/webroot

* [Check the documentation for NFS on macOS here](https://github.com/reload/proref/tree/feature/docker-for-mac-nfs/docker-for-mac/nfs)

### CODE REVIEW

File: drupal8-example/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+up:
```
Comment (arnested) [inline drupal8-example/Makefile]: Why introduce the Makefile in this PR?

There is a Makefile example in another folder of the Git root. That might no be desirable in the long run but is not really related to the NFS stuff(?)

File: drupal8-example/Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	@docker-compose up -d
+endif
+
+	# Tada!! 🎉 Run 'make logs' to follow output.
+
+down:
+ifeq ($(shell uname), Darwin)
+	@echo "Looks like you're on macOS! Using docker-compose-mac.yml"
+	@docker-compose -f docker-compose.yml -f docker-compose-mac.yml stop
+	@docker-compose -f docker-compose.yml -f docker-compose-mac.yml down -v
```
Comment (arnested) [inline drupal8-example/Makefile]: `--remove-orphans` is not necessary on Mac?

File: drupal8-example/Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+up:
+ifeq ($(shell uname), Darwin)
+	@echo "Looks like you're on macOS! Using docker-compose-mac.yml"
+	@docker-compose -f docker-compose.yml -f docker-compose-mac.yml up -d
```
Comment (arnested) [inline drupal8-example/Makefile]: Why hide what command is actually run?

Transparency into what magic is actually happening behind the scenes seems like a good idea (and could even remove the need for the echo above).

**Comment by tripox** [comment]
The `usePolling` options needs to be `true` when using NFS: https://gulpjs.org/API.html#opts
This introduces no problems or performance issues for Linux but is required on macOS with NFS in order for Gulp to watch for file changes.

Any suggestions to where to put this information? @arnested?

File: drupal8-example/Makefile (hunk 3 above)
Comment (tripox) [reply drupal8-example/Makefile]: Good point. I'll fix it.

File: drupal8-example/Makefile (hunk 2 above)
Comment (tripox) [reply drupal8-example/Makefile]: I've added it.

File: drupal8-example/Makefile [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+ifeq ($(shell uname), Darwin)
+	docker-compose -f docker-compose.yml -f docker-compose-mac.yml stop
+	docker-compose -f docker-compose.yml -f docker-compose-mac.yml down -v --remove-orphans
+else
+	docker-compose stop
+	docker-compose down -v --remove-orphans
+endif
+
+logs:
+	@docker-compose logs -f
```
Comment (arnested) [inline drupal8-example/Makefile]: Also why hide the command here?

File: drupal8-example/Makefile (hunk 1 above)
Comment (tripox) [reply drupal8-example/Makefile]: Ah, didn't see that one. :) 
It is related to the NFS stuff however (we go from bind volumes to NFS volumes on macOS)

**Comment by xendk** [comment]
How about incorporating the changes I made to https://github.com/reload/verdensbiblioteket.dk/blob/master/Makefile ?

File: drupal8-example/Makefile (hunk 4 above)
Comment (arnested) [reply drupal8-example/Makefile]: In a couple of months someone will add a service only for the mac setup. That services logs will not tailed here.

Have you considered moving the OS detecting out the individual targets and instead do it once and have it set just [COMPOSE_FILE environment file](https://docs.docker.com/compose/reference/envvars/#compose_file). It believe that could get us more readable, more maintainable,  and less error prone Makefiles.

File: drupal8-example/Makefile (hunk 1 above)
Comment (arnested) [reply drupal8-example/Makefile]: Yeah, but having two different Makefile examples around just leads to incompatibility.

File: drupal8-example/Makefile (hunk 4 above)
Comment (tripox) [reply drupal8-example/Makefile]: That's a good idea!

File: drupal8-example/Makefile (hunk 1 above)
Comment (tripox) [reply drupal8-example/Makefile]: I agree! I'm removing it. :)

File: drupal8-example/Makefile (hunk 4 above)
Comment (arnested) [reply drupal8-example/Makefile]: Remember variables in Makedfiles are not exported by default so you'll need to do that explicitly:
```
export COMPOSE_FILE = docker-compose.yml:docker-compose-mac.yml
```
(also remember to consider impacts on the `docker-compose.override.yml`)

Comment (tripox) [reply drupal8-example/Makefile]: Fixed

Comment (arnested) [reply drupal8-example/Makefile]: Fixed? No, it's not. You took the idea from this approach and added it to the docker reset script(s). But, you did not use the approach when you moved the functionality into the other Makefile example.

**Pull request closed without merging**